openapi/themoviedb/client.bal

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import  ballerina/http;
+import  ballerina/url;
+import  ballerina/lang.'string;
+
+# Please visit [here](https://wwww.themoviedb.org/) and obtain an `API Key` in the settings.
+#
+# + apiKeys - Provide your API Key as `api_key`. Eg: `{"api_key" : "<Your API Key>"}` 
+public type ApiKeysConfig record {
+    map<string|string[]> apiKeys;
+};
+
+# Get popular movies response
+public type GetPopularMoviesResponse record {
+    # Specify which page to query
+    int page;
+    # Movie List Object
+    MovieListObject[] results;
+    # Total number of results
+    int total_results;
+    # Total number of pages
+    int total_pages;
+};
+
+# Get upcoming movies response
+public type GetUpcomingMoviesResponse record {
+    # Specify which page to query
+    int page;
+    # Movie List Object
+    MovieListObject[] results;
+    # Release date range of movies
+    record  { string maximum?; string minimum?;}  dates;
+    # Total number of pages
+    int total_pages;
+    # Total number of results
+    anydata total_results;
+};
+
+# Get movie by movie ID response
+public type GetMovieByMovieIdResponse record {
+    # Adult content or not
+    boolean adult?;
+    # Backdrop image path
+    string? backdrop_path?;
+    # Budget of movie
+    int budget?;
+    # Genres of movie
+    record  { int id?; string name?;} [] genres?;
+    # Movie ID
+    int id;
+    # Original language of movie
+    string original_language?;
+    # Original title of movie
+    string original_title?;
+    # Popularity of movie
+    decimal popularity?;
+    # Poster image path
+    string? poster_path?;
+    # Movie production companies
+    record  { string name?; int id?; string origin_country?;} [] production_companies?;
```

Comment:
I think this happens because of the defined schemas that have inline schemas. We had better make them defined schemas before releasing the connector. otherwise, there might be issues in choreo connector parsing.

Reply:
Lets handle those inline schema to create nested records by adding them in openapi specification